src/app/modules/submission/submission.service.ts

```diff
@@ -251,17 +258,64 @@ export const copyPendingSubmissionToSubmissions = (
       return okAsync(submission)
     })
     .andThen((pendingSubmission) => {
-      const submission = new SubmissionModel(
-        omit(pendingSubmission, ['_id', 'created', 'lastModified']),
-      )
+      const submissionContent = omit(pendingSubmission, [
+        '_id',
+        'created',
+        'lastModified',
+      ])
+      const submission = new SubmissionModel({
+        // Explicitly copy over the pending submission's _id
+        ...submissionContent,
+        _id: pendingSubmissionId,
+      })
 
       return ResultAsync.fromPromise(submission.save({ session }), (error) => {
         logger.error({
           message: 'Database save submission error',
           meta: logMeta,
           error,
         })
-        return new DatabaseError(getMongoErrorMessage(error))
+
+        const databaseError = transformMongoError(error)
+
+        return databaseError instanceof DatabaseDuplicateKeyError
+          ? databaseError
+          : new DatabaseError(getMongoErrorMessage(error))
+      }).orElse((error) => {
```

Comment:
Could you add tests for the retrying logic? Seems like its a pretty important mechanism for payments that we will not want to break with future code changes.

Reply:
done in [44796fa](https://github.com/opengovsg/FormSG/pull/6088/commits/44796fa5173cfd6188b0a76d7e4460a39f8f403c)